data-processor/src/main/java/io/micronaut/data/processor/visitors/RepositoryTypeElementVisitor.java

```diff
@@ -252,6 +254,28 @@ public void visitMethod(MethodElement element, VisitorContext context) {
                         entityResolver
                 );
 
+                // choose right join alias
+                // ignore pageable methods because they don't support join aliases
+                if (element.hasAnnotation(JoinSpecifications.class) && !methodMatchContext.hasParameterInRole(TypeRole.PAGEABLE)) {
```

Comment:
The logic should be implemented in `AbstractSqlLikeQueryBuilder`: `getAliasName` or `getPathOnlyAliasName`.